src/Events/QR/Shortcode.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * The Shortcode class for the QR module.
+ *
+ * @since TBD
+ */
+
+namespace TEC\Events\QR;
+
+use Tribe\Shortcode\Shortcode_Abstract;
+use TEC\Common\QR\QR;
+
+/**
+ * Class Shortcode
+ *
+ * @since TBD
+ *
+ * @package TEC\Events\QR
+ */
+class Shortcode extends Shortcode_Abstract {
+	/**
+	 * The shortcode tag.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected $slug = 'tec_event_qr';
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @var array
+	 */
+	protected $default_arguments = [
+		'mode' => 'current',
+		'id'   => '',
+		'size' => 6,
+	];
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @var array
+	 */
+	public $validate_arguments_map = [
+		'id'   => 'tribe_post_exists',
+		'mode' => 'sanitize_title_with_dashes',
+		'size' => 'absint',
+	];
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @since TBD
+	 *
+	 * @return string
+	 */
+	public function get_html() {
+
+		$args = $this->get_arguments();
+
+		$mode = in_array( $args['mode'], [ 'current', 'next', 'id', 'series_next' ], true ) ? $args['mode'] : 'current';
```

Comment:
It's better to avoid using `in_array()`, as that could require going through every array element instead of checking only one time.

```suggestion
		$valid_modes = [
			'current' => true,
			'next' => true,
			'id' => true,
			'series_next' => true,
		];
		$mode = isset( $valid_modes[ $args['mode'] ] ? $args['mode'] : 'current';
```